App.js

```diff
@@ -306,25 +313,41 @@ export default class App extends React.Component {
             linking={linking}
             theme={navTheme}
             onStateChange={(state) => {
-              console.log('STATE CHANGED: ', state);
+              const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
+              const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
+              const isRHPOnTopOfStack = lastRHPIndex === state.index;
+              this.stateRef.current = {
+                ...this.stateRef.current,
+                state,
+                lastChatIndex,
+                lastRHPIndex,
+                isRHPOnTopOfStack
```

Comment:
Can all of these other values be derived just from looking at the `state`? I'm not sure I get why we need to store them all.